crhchttp/http_cloud_dot_redhat.go

```diff
@@ -34,7 +35,14 @@ var log = logr.Log.WithName("crc_http")
 
 // DefaultTransport is a copy from the golang http package
 var DefaultTransport = &http.Transport{
-	Proxy: http.ProxyFromEnvironment,
+	// TODO: remove conditional proxy if not necessary going forward
+	Proxy: func(req *http.Request) (*url.URL, error) {
+		env := os.Getenv("API_TARGET_ENV")
+		if env == "stage" {
+			return url.Parse("http://squid.corp.redhat.com:3128")
+		}
+		return http.ProxyFromEnvironment(req)
+	},
```

Comment:
This can also be added in a separate PR. Leaving it here for now to get through testing in stage.